File: Questions/Java/Pair_sumfast.java
import java.util.*;

class Pair_sumfast
{
    public static void main(String[] args)
    {
        Scanner sc = new Scanner(System.in);

        int n= sc.nextInt();
        int k= sc.nextInt();

        int[] arr= new int[n];
        arr[0] = sc.nextInt();
        int max = arr[0];
        for(int i =1 ;i < n ; i++)
        {
            arr[i]= sc.nextInt();
            if(max<arr[i])
                max = arr[i];
        }
        int[] v= new int[max+1];
        Arrays.fill(v,0);
        int flag =0;
        for(int i = 0; i < n; i++)
        {
            int t = k - arr[i];
            
                if(v[t] == 1)
                {
                    flag = 1;
                    break;
                }
                v[arr[i]]=1;
        }
        if(flag == 0)
            System.out.println("NO");
        else 
            System.out.println("YES");
    }
}

File: Questions/cpp/pair_sum_fast.cpp
#include<bits/stdc++.h>
using namespace std;
 
int main()
{
    vector <bool> v(1000000,0);
    int n,k;
    cin>>n>>k;
    int arr[n];
    int flag = 0;
    for(int i = 0; i < n ; i++)
    {
        cin>>arr[i];
       
    }
    for(int i = 0; i < n; i++)
    {
        int t = k - arr[i];
        
            if(v[t] == 1)
            {
                flag = 1;
                break;
            }
            v[arr[i]]=1;
            
    }
    if(flag == 0)
        cout<<"NO";
    else 
        cout<<"YES";
        
    
}